docs/usecases/testing.md

```diff
@@ -72,3 +72,271 @@ Test aspects are powerful tools for modifying behavior of individual tests or ev
 **Zero Dependencies**
 
 As a library with zero third party dependencies, this project is available on the JVM, ScalaJS, and will be available on Dotty and Scala Native in the near future. So you can write your tests once and make sure that your code works correctly across all platforms that you support.
+
+
+**How zio-test was designed**
+
+`zio-test` is designed around the idea of making test first-class objects. What it means is that tests 
+(and other accompanying concepts like assertions) become ordinary values that can be passed around, transformed and composed together. This approach allows for greater flexibility comparing to some other testing frameworks where tests and additional logic around tests had to be put into callbacks so that framework could make use of them. This approach also fits better with other `ZIO` concepts like `ZManaged` which can only be used within a scoped block of code. This also created a mismatch between `BeforeAll`, `AfterAll` callback-like methods when there were resources that should be opened and closed during test suite execution.
+
+_Constructing tests_
+
+All below code assumes that you have imported `zio.test._`
+
+The backbone of `zio-test` is the `Spec[L, T]` class. Every spec is labeled with `L` and can be a suite which contains other specs or a test of type `T`.
+
+The most common and easy way to create suits and tests is to use the `suite` and `testM` functions. 
+
+```scala
+import zio.test._
+import zio.clock.nanoTime
+import Assertion.isGreaterThan
+
+val clockSuite = suite("clock") {
+  testM("time is non-zero") {
+    assertM(nanoTime, isGreaterThan(0))
+  }
+}
+```
+
+As you can see the whole suit was assigned to `clockSuite` val. As it was said suits can contain other suits so we can aggregate them as much as needed. Example, we can have multiple suits that test external HTTP apis and one big suit that will aggregate them all.
+
+```scala
+val paymentProviderABCSuite  = suit("ABC payment provider tests") {/* tests */}
+val paymentProviderXYZSuite  = suit("XYZ payment provider tests") {/* tests */}
+val allPaymentProvidersTests = suit("All payment providers tests")(paymentProviderABCSuite, paymentProviderXYZSuite)
+```
+
+Real tests that run some logic and return testing result are created mostly with `testM` function. It expects two arguments, first one that will be the label of test
+which will be used for visual reporting back to the user and an assertion of type
+`ZIO[R, E, TestResult]`. This means writing test in `zio-test` mostly gets down to creating a `ZIO` object that 
+will produce `TestResult`. There is another variant of function for creating test that are pure called simply `test`.
+It expects a thunk of code that will just return a `TestResult` without packing it into `ZIO`.
+
+_Assertions - creating TestResults_
+As it was already mentioned tests should return `TestResult`. The most common way to produce a `TestResult`
```

Comment:
I think this is really good. The other thing I might add here is how a lot of assertions compose by taking other assertions as parameters. For example you could write `isRight(isSome(equalTo(2)))`. The idea with this and using methods like `&&` and `||` is that a lot of examples for testing frameworks have these really simply tests where you are comparing two integers or something, so if the test fails it is obvious what went wrong. But in real life maybe you are comparing part of two large case classes, or a bunch of different statements about some test. So it isn't that helpful to just say this test failed. We want to really zero in for you on what part of your test failed so it is easy for you to diagnose the problem and fix it.

Reply:
👍